engine.py

```diff
@@ -134,8 +134,13 @@ def post_app_init(self):
             self.log_debug("Updating the Work Area on the current project to '%s'" % self.context)
             self.__metadata_mgr.set_project_metadata(current_project, self.context)
 
-        # connect to Mari project events:
+
+        # connect to Mari project events
+        # try disconnecting initially because this seems to be called multiple times
+        mari.utils.disconnect(mari.projects.opened, self.__on_project_opened)
+        mari.utils.disconnect(mari.projects.saved, self.__on_project_saved)
```

Comment:
There is no function in mari to check what callbacks are already connected to a signal...but disconnecting just returns False if the connection doesn't exist. Seems safe to do this.
__on_project_opened() just tries to switch context. Execution of this multiple times is avoided by checking if current and changed context are the same. But we cannot do that for file save (we want to overwrite even if file exists, because this is a WIP save)